--- ../src-base/minecraft/net/minecraft/client/renderer/BlockRendererDispatcher.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/BlockRendererDispatcher.java
@@ -1,5 +1,6 @@
 package net.minecraft.client.renderer;
 
+import java.util.ArrayList;
 import net.minecraft.block.Block;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.client.renderer.texture.TextureAtlasSprite;
@@ -11,11 +12,13 @@
 import net.minecraft.client.settings.GameSettings;
 import net.minecraft.crash.CrashReport;
 import net.minecraft.crash.CrashReportCategory;
+import net.minecraft.init.Blocks;
 import net.minecraft.util.BlockPos;
 import net.minecraft.util.MathHelper;
 import net.minecraft.util.ReportedException;
 import net.minecraft.world.IBlockAccess;
 import net.minecraft.world.WorldType;
+import net.minecraftforge.client.ITessellatorRenderer;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
 
@@ -33,6 +36,11 @@
     {
         this.field_175028_a = p_i46237_1_;
         this.field_175026_b = p_i46237_2_;
+        // Clumsy.
+        Blocks.field_150358_i.setTessellatorRenderer(field_175025_e);
+        Blocks.field_150355_j.setTessellatorRenderer(field_175025_e);
+        Blocks.field_150356_k.setTessellatorRenderer(field_175025_e);
+        Blocks.field_150353_l.setTessellatorRenderer(field_175025_e);
     }
 
     public BlockModelShapes func_175023_a()
@@ -52,6 +60,19 @@
             IBakedModel ibakedmodel1 = (new SimpleBakedModel.Builder(ibakedmodel, p_175020_3_)).func_177645_b();
             this.field_175027_c.func_178259_a(p_175020_4_, ibakedmodel1, p_175020_1_, p_175020_2_, Tessellator.func_178181_a().func_178180_c());
         }
+        else if (i == 1)
+        {
+            block.getTessellatorRenderer().renderDamage(p_175020_4_, p_175020_1_, p_175020_2_, Tessellator.func_178181_a().func_178180_c());
+        }
+        else if (i == 4)
+        {
+            WorldRenderer wr = Tessellator.func_178181_a().func_178180_c();
+            IBakedModel[] segments = block.getSegmentedModel().getModels(p_175020_4_, p_175020_2_, p_175020_1_);
+            for (IBakedModel model: segments)
+            {
+                this.field_175027_c.func_178259_a(p_175020_4_, model, p_175020_1_, p_175020_2_, wr);
+            }
+        }
     }
 
     public boolean func_175018_a(IBlockState p_175018_1_, BlockPos p_175018_2_, IBlockAccess p_175018_3_, WorldRenderer p_175018_4_)
@@ -69,12 +90,20 @@
                 switch (i)
                 {
                     case 1:
-                        return this.field_175025_e.func_178270_a(p_175018_3_, p_175018_1_, p_175018_2_, p_175018_4_);
+                        return p_175018_1_.func_177230_c().getTessellatorRenderer().func_178270_a(p_175018_3_, p_175018_1_, p_175018_2_, p_175018_4_);
                     case 2:
                         return false;
                     case 3:
                         IBakedModel ibakedmodel = this.func_175022_a(p_175018_1_, p_175018_3_, p_175018_2_);
                         return this.field_175027_c.func_178259_a(p_175018_3_, ibakedmodel, p_175018_1_, p_175018_2_, p_175018_4_);
+                    case 4:
+                        boolean any = false;
+                        IBakedModel[] segments = p_175018_1_.func_177230_c().getSegmentedModel().getModels(p_175018_3_, p_175018_2_, p_175018_1_);
+                        for (IBakedModel model : segments)
+                        {
+                            any |= this.field_175027_c.func_178259_a(p_175018_3_, model, p_175018_1_, p_175018_2_, p_175018_4_);
+                        }
+                        return any;
                     default:
                         return false;
                 }
@@ -142,6 +171,7 @@
             {
                 case 1:
                 default:
+                    p_175016_1_.func_177230_c().getTessellatorRenderer().setModelBrightness(p_175016_1_, p_175016_2_);
                     break;
                 case 2:
                     this.field_175024_d.func_178175_a(p_175016_1_.func_177230_c(), p_175016_2_);
@@ -169,5 +199,17 @@
     public void func_110549_a(IResourceManager p_110549_1_)
     {
         this.field_175025_e.func_178268_a();
+        for (ITessellatorRenderer renderer : registeredRenderers)
+        {
+            renderer.func_178268_a();
+        }
+        // Do segmented models need notification of resourcepack-reload?
     }
+    
+    public void registerTessellatorRenderer(ITessellatorRenderer renderer)
+    {
+        registeredRenderers.add(renderer);
+    }
+    
+    private ArrayList<ITessellatorRenderer> registeredRenderers = new ArrayList();
 }
