--- ../src-base/minecraft/net/minecraft/client/renderer/BlockFluidRenderer.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/BlockFluidRenderer.java
@@ -12,9 +12,10 @@
 import net.minecraft.world.IBlockAccess;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
+import net.minecraftforge.client.ITessellatorRenderer;
 
 @SideOnly(Side.CLIENT)
-public class BlockFluidRenderer
+public class BlockFluidRenderer implements ITessellatorRenderer
 {
     private TextureAtlasSprite[] field_178272_a = new TextureAtlasSprite[2];
     private TextureAtlasSprite[] field_178271_b = new TextureAtlasSprite[2];
@@ -287,4 +288,17 @@
 
         return 1.0F - f / (float)i;
     }
+    
+    @Override
+    public void setModelBrightness(IBlockState state, float brightness)
+    {
+        // Unused
+    }
+    
+    @Override
+    public void renderDamage(IBlockAccess world, IBlockState state, BlockPos position, WorldRenderer worldRenderer)
+    {
+        // Unused
+    }
+    
 }
