--- ../src-base/minecraft/net/minecraft/client/renderer/BlockFluidRenderer.java
+++ ../src-work/minecraft/net/minecraft/client/renderer/BlockFluidRenderer.java
@@ -12,9 +12,10 @@
 import net.minecraft.world.IBlockAccess;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
+import net.minecraftforge.client.ITessellatorRenderer;
 
 @SideOnly(Side.CLIENT)
-public class BlockFluidRenderer
+public class BlockFluidRenderer implements ITessellatorRenderer
 {
     private TextureAtlasSprite[] field_178272_a = new TextureAtlasSprite[2];
     private TextureAtlasSprite[] field_178271_b = new TextureAtlasSprite[2];
@@ -25,7 +26,7 @@
         this.func_178268_a();
     }
 
-    protected void func_178268_a()
+    public void func_178268_a()
     {
         TextureMap texturemap = Minecraft.func_71410_x().func_147117_R();
         this.field_178272_a[0] = texturemap.func_110572_b("minecraft:blocks/lava_still");
@@ -287,4 +288,17 @@
 
         return 1.0F - f / (float)i;
     }
+    
+    @Override
+    public void setModelBrightness(IBlockState state, float brightness)
+    {
+        // Unused
+    }
+    
+    @Override
+    public void renderDamage(IBlockAccess world, IBlockState state, BlockPos position, WorldRenderer worldRenderer)
+    {
+        // Unused
+    }
+    
 }
